src/meta/src/stream/scheduler.rs

```diff
@@ -222,12 +223,14 @@ impl Scheduler {
 
             // Record actor locations and set vnodes into the actors.
             for (actor, parallel_unit) in fragment.actors.iter_mut().zip_eq(parallel_units) {
-                let parallel_unit =
-                    if actor.same_worker_node_as_upstream && !actor.upstream_actor_id.is_empty() {
-                        locations.schedule_colocate_with(&actor.upstream_actor_id)?
-                    } else {
-                        parallel_unit.clone()
-                    };
+                let parallel_unit = if let Some(colocated_actor_id) = &actor.colocated_upstream_actor_id
+                    && !actor.upstream_actor_id.is_empty()
+                {
+                    locations
+                        .schedule_colocate_with(&[colocated_actor_id.id])?
```

Comment:
Do we ensure that the colocated actor is already scheduled at this time?

Reply:
In order not to worry about such trivial issues, let us put our hopes in the new scheduler. 😆